overlaysPackagesBuilder.nix

```diff
@@ -0,0 +1,106 @@
+{ flake-utils-plus }: let
+
+ overlaysPackagesBuilder = pkgs: channels:
+  /**
+  Synopsis: overlaysPackagesBuilder _pkgs_ _channels_
+
+  pkgs:     pkgs.<system>.<tree>
+  channels: channels.<name>.overlays
+
+  Returns valid packges that have been defined within an overlay so they
+  can be shared via self.packages with the world. This is especially useful
+  over sharing one's art via self.overlays in case you have a binary cache
+  running from which third parties could benefit.
+
+  First, flattens an arbitrarily nested _pkgs_ tree for each system into a flat
+  key in which nesting is aproximated by a "/" (e.g. "development/kakoune").
+  Also filter the resulting packages for attributes that _trivially_ would
+  fail flake checks (broken, system not supported or not a derivation).
+
+  Second, collects all overlays' packages' keys of all channels into a flat list.
+
+  Finally, only passes packages through the seive that are prefixed with a top level
+  key exposed by any of the overlays. Since overlays override (and do not emrge)
+  top level attributes, by filtering on the prefix, an overlay's entire packages
+  tree will be correctly captured.
+
+  Example:
+
+  pkgs'.<system> = {
+    "development/kakoune" = { ... };
```

Comment:
To be clear this means that these packages were put into pkgs as `pkgs.development.kakoune` right?

Reply:
Correct, that is: the overlay key was named (for whatever reason) "development" and had an attribute set marked with `{ recurseForDerivations = true; }`

Typically that would have to look like so:

```nix
{
final: prev: {
  development = prev.recurseIntoAttrs {
    kakoune = {};
  };
}
}
```

If not setup this way, flattenTree would not recurse into the attribute set (just as any other standard nixpkgs tool in similar situations).

Reply:
ok perfect thanks for the explanation. That makes a lot of sense.